typescript/web/src/components/spinner/index.tsx

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { Spinner as ChakraSpinner, ThemingProps } from "@chakra-ui/react";
+
+type SpinnerProps = {
+  id?: string | undefined;
+  color?: string | undefined;
+  ariaLabel?: string | undefined;
+  size?: ThemingProps<"Spinner">["size"] | undefined;
+  visibility?: "visible" | "hidden" | undefined;
+  key?: string | undefined;
+  ref?: React.Ref<HTMLDivElement> | undefined;
+};
```

Comment:
Why don't you directly reuse `SpinnerProps` from Chakra?

Reply:
Done